src/app/pages/HomePage/providers/MobileTooltipProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import { createContext, FC, PropsWithChildren, useContext, useState } from 'react'
+import { GraphEndpoint } from '../ParaTimeSelector/Graph/types'
+
+interface MobileTooltipProviderState {
+  showMobileTooltip: { [key in GraphEndpoint]: boolean }
+}
+
+interface MobileTooltipProviderContext {
+  readonly state: MobileTooltipProviderState
+  setShowMobileTooltip: (graphEndpoint: GraphEndpoint, show: boolean) => void
+}
+
+const mobileTooltipProviderInitialState: MobileTooltipProviderState = {
+  showMobileTooltip: {
+    [GraphEndpoint.Consensus]: false,
+    [GraphEndpoint.Emerald]: false,
+    [GraphEndpoint.Sapphire]: false,
+    [GraphEndpoint.Cipher]: false,
+  },
+}
+
+export const MobileTooltipContext = createContext<MobileTooltipProviderContext>(
+  {} as MobileTooltipProviderContext,
+)
+
+export const MobileTooltipProvider: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
Not sure how we feel about `Context` usage, but it was better solution than prop drilling.

Reply:
useMobileTooltip is used in HomePage. It is also used in ParaTimeSelector which is a children of HomePage. When prop drilling started to being a problem here ?

Reply:
Right, so state is simplified now. The problem being that this is being passed 2 level deep. I can move to simple `useState`, if it makes more sense.